oc/openshift_ops.py

```diff
@@ -184,3 +185,165 @@ def create_project(self, project):
+            str : name of the cephblockpool created
+
+        Raises:
+            KeyError when error occured
+
+        Examples:
+            create_cephblockpool(
+                namespace,
+                cephblockpool_name_prefix="autotests-blockpool",
+                failureDomain="host",
+                size=3)
+
+        """
+        service_cbp = self.dyn_client.resources.get(
+            api_version='v1',
+            kind='CephBlockPool'
+        )
+        cephblockpool_name = "%s-%s" % (
+            cephblockpool_name_prefix,
+            get_random_str()
+        )
+        # overwrite the namespace with openshift-storage, since cephblockpool
+        # is tied-up with openshift-storage
+        namespace = "openshift-storage"
+        cephblockpool_data = {}
+        cephblockpool_data['cephblockpool_name'] = cephblockpool_name
+        cephblockpool_data['rook_api_version'] = defaults.rook_api_version
+        cephblockpool_data['failureDomain'] = kwargs.get('failureDomain')
+        cephblockpool_data['size'] = kwargs.get('size')
+
+        data = generate_data("cephblockpool.yaml", cephblockpool_data)
+        service_cbp.create(body=data, namespace=namespace)
+
+        return cephblockpool_name
+
+    def create_storageclass(
+        self,
+        sc_name_prefix="autotests-sc",
+        allow_volume_expansion=True,
+        reclaim_policy="Delete",
+        **kwargs
+    ):
+        """
+        Creates storage class using data provided
+
+        Args:
+            sc_name_prefix (str): sc name will consist of this prefix and
+                                  random str.
+
+        Kwargs:
+            All the keyword arguments are expected to be key and values of
+            'kwargs' section for storage class.
+
+        Returns:
+            str: name of the storage class created
+
+        Example:
+            create_storageclass(
+                sc_name_prefix="autotests-sc",
+                allow_volume_expansion=True,
+                reclaim_policy="Delete",
+                blockPool=blockPool,
+                clusternamespace="openshift-storage",
+                fstype="xfs"
+            )
+
+        """
+        service_sc = self.dyn_client.resources.get(
+            api_version='v1',
+            kind='StorageClass'
+        )
+        storageclass_name = "%s-%s" % (sc_name_prefix, get_random_str())
+
+        sc_data = {}
+        sc_data['k8s_api_version'] = defaults.k8s_api_version
+        sc_data['storageclass_name'] = storageclass_name
+        sc_data['volume_expansion'] = allow_volume_expansion
+        sc_data['reclaimPolicy'] = reclaim_policy
+        sc_data['blockPool'] = kwargs.get('blockPool')
+        sc_data['clusterNamespace'] = kwargs.get('clusterNamespace')
+        sc_data['fstype'] = kwargs.get('fstype')
+
+        data = generate_data("storageclass.yaml", sc_data)
+        service_sc.create(body=data)
+
+        return storageclass_name
+
+    def create_pvc(
```

Comment:
Let's move this to another class VPC

Reply:
+1 here. I see a more natural heirarchy of kubernetes volumes . Class KubeVolume() and child classes could be PVC(KubeVolume), RBD(KubeVolume), CephFS(KubeVolume). I haven't really worked on all of these from kubernetes perspective but by looking at kubernetes docs this is what I felt. May be other can shine some light on this.

Reply:
Implemented the same in volume.py